File: MiniDOMBase/DMD/Sistema/Calendar/Activities/CCalendarActivityInfo.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica

Partial Class Sistema

    ''' <summary>
    ''' Racchiude alcune info sui contatti
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CCalendarActivityInfo
        Implements XML.IDMDXMLSerializable

        Public IDRicontatto As Integer
        Public TipoRicontatto As String
        Public IDContattoPrecedente As Integer
        Public DataProssimoRicontatto As Date?
        Public DataProssimoAppuntamento As Date?
        Public DataUltimoRicontatto As Date?
        Public DataUltimoAppuntamento As Date?
        Public NumeroPraticheInCorso As Integer
        Public NumeroPraticheArchiviate As Integer

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
            IDRicontatto = 0
            TipoRicontatto = ""
            IDContattoPrecedente = 0
            DataProssimoRicontatto = Nothing
            DataProssimoAppuntamento = Nothing
            DataUltimoRicontatto = Nothing
            DataUltimoAppuntamento = Nothing
            NumeroPraticheInCorso = 0
            NumeroPraticheArchiviate = 0
        End Sub

        Public Sub Initialize(ByVal personID As Integer)
            'Dim cursor As New CCustomerCallsCursor
            'cursor.IDPersona.Value = personID
            'cursor.Stato.Value = ObjectStatus.OBJECT_VALID
            ''cursor.Ricontattare.Value = True
            ''cursor.DataRicontatto.SortOrder = SortEnum.SORT_ASC
            'Me.IDContattoPrecedente = GetID(cursor.Item)
            '' If Not (cursor.Item Is Nothing) Then Me.DataProssimoRicontatto = cursor.Item.DataRicontatto
            'cursor.Reset()

            'cursor = New CCustomerCallsCursor
            'cursor.IDPersona.Value = personID
            'cursor.Stato.Value = ObjectStatus.OBJECT_VALID
            'cursor.Data.SortOrder = SortEnum.SORT_DESC

            'If Not (cursor.Item Is Nothing) Then Me.DataUltimoRicontatto = cursor.Item.Data
            'cursor.Reset()

            'Me.IDRicontatto = GetID(Ricontatti.GetProssimoRicontatto(personID))
        End Sub

        Protected Sub XMLSerialize(ByVal writer As minidom.XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("IDRicontatto", Me.IDRicontatto)
            writer.WriteAttribute("TipoRicontatto", Me.TipoRicontatto)
            writer.WriteAttribute("IDContattoPrecedente", Me.IDContattoPrecedente)
            writer.WriteAttribute("DataProssimoRicontatto", Me.DataProssimoRicontatto)
            writer.WriteAttribute("DataProssimoAppuntamento", Me.DataProssimoAppuntamento)
            writer.WriteAttribute("DataUltimoRicontatto", Me.DataUltimoRicontatto)
            writer.WriteAttribute("DataUltimoAppuntamento", Me.DataUltimoAppuntamento)
            writer.WriteAttribute("NumeroPraticheInCorso", Me.NumeroPraticheInCorso)
            writer.WriteAttribute("NumeroPraticheArchiviate", Me.NumeroPraticheArchiviate)
        End Sub

        Protected Sub SetFieldInternal(ByVal fieldName As String, ByVal fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "IDRicontatto" : Me.IDRicontatto = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "TipoRicontatto" : Me.TipoRicontatto = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "IDContattoPrecedente" : Me.IDContattoPrecedente = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "DataProssimoRicontatto" : Me.DataProssimoRicontatto = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "DataProssimoAppuntamento" : Me.DataProssimoAppuntamento = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "DataUltimoRicontatto" : Me.DataUltimoRicontatto = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "DataUltimoAppuntamento" : Me.DataUltimoAppuntamento = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "NumeroPraticheInCorso" : Me.NumeroPraticheInCorso = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "NumeroPraticheArchiviate" : Me.NumeroPraticheArchiviate = XML.Utils.Serializer.DeserializeInteger(fieldValue)
            End Select
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class


End Class

File: MiniDOMBase/DMD/Sistema/Calendar/Activities/Info/CCalendarActivityInfo.cs
using System;
using DMD;
using DMD.XML;

namespace minidom
{
    public partial class Sistema
    {

        /// <summary>
        /// Racchiude alcune info sui contatti
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CCalendarActivityInfo 
            : IDMDXMLSerializable
        {
            /// <summary>
            /// ID ricontatto
            /// </summary>
            public int IDRicontatto;

            /// <summary>
            /// TipoRicontatto
            /// </summary>
            public string TipoRicontatto;

            /// <summary>
            /// IDContattoPrecedente
            /// </summary>
            public int IDContattoPrecedente;

            /// <summary>
            /// DataProssimoRicontatto
            /// </summary>
            public DateTime? DataProssimoRicontatto;

            /// <summary>
            /// DataProssimoAppuntamento
            /// </summary>
            public DateTime? DataProssimoAppuntamento;

            /// <summary>
            /// DataUltimoRicontatto
            /// </summary>
            public DateTime? DataUltimoRicontatto;

            /// <summary>
            /// DataUltimoAppuntamento
            /// </summary>
            public DateTime? DataUltimoAppuntamento;

            /// <summary>
            /// NumeroPraticheInCorso
            /// </summary>
            public int NumeroPraticheInCorso;

            /// <summary>
            /// NumeroPraticheArchiviate
            /// </summary>
            public int NumeroPraticheArchiviate;

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCalendarActivityInfo()
            {
                //DMDObject.IncreaseCounter(this);
                IDRicontatto = 0;
                TipoRicontatto = "";
                IDContattoPrecedente = 0;
                DataProssimoRicontatto = default;
                DataProssimoAppuntamento = default;
                DataUltimoRicontatto = default;
                DataUltimoAppuntamento = default;
                NumeroPraticheInCorso = 0;
                NumeroPraticheArchiviate = 0;
            }

            void IDMDXMLSerializable.XMLSerialize(XMLWriter writer)
            {
                this.XMLSerialize(writer);
            }

            void IDMDXMLSerializable.SetFieldInternal(string fieldName, object fieldValue)
            {
                this.SetFieldInternal(fieldName, fieldValue);
            }

            /// <summary>
            /// Inizializza
            /// </summary>
            /// <param name="personID"></param>
            public void Initialize(int personID)
            {
                // Dim cursor As New CContattoUtenteCursor
                // cursor.IDPersona.Value = personID
                // cursor.Stato.Value = ObjectStatus.OBJECT_VALID
                // 'cursor.Ricontattare.Value = True
                // 'cursor.DataRicontatto.SortOrder = SortEnum.SORT_ASC
                // Me.IDContattoPrecedente = GetID(cursor.Item)
                // ' If Not (cursor.Item Is Nothing) Then Me.DataProssimoRicontatto = cursor.Item.DataRicontatto
                // cursor.Reset()

                // cursor = New CContattoUtenteCursor
                // cursor.IDPersona.Value = personID
                // cursor.Stato.Value = ObjectStatus.OBJECT_VALID
                // cursor.Data.SortOrder = SortEnum.SORT_DESC

                // If Not (cursor.Item Is Nothing) Then Me.DataUltimoRicontatto = cursor.Item.Data
                // cursor.Reset()

                // Me.IDRicontatto = GetID(Ricontatti.GetProssimoRicontatto(personID))
            }

            /// <summary>
            /// Serializzazione xml
            /// </summary>
            /// <param name="writer"></param>
            protected void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("IDRicontatto", IDRicontatto);
                writer.WriteAttribute("TipoRicontatto", TipoRicontatto);
                writer.WriteAttribute("IDContattoPrecedente", IDContattoPrecedente);
                writer.WriteAttribute("DataProssimoRicontatto", DataProssimoRicontatto);
                writer.WriteAttribute("DataProssimoAppuntamento", DataProssimoAppuntamento);
                writer.WriteAttribute("DataUltimoRicontatto", DataUltimoRicontatto);
                writer.WriteAttribute("DataUltimoAppuntamento", DataUltimoAppuntamento);
                writer.WriteAttribute("NumeroPraticheInCorso", NumeroPraticheInCorso);
                writer.WriteAttribute("NumeroPraticheArchiviate", NumeroPraticheArchiviate);
            }

            /// <summary>
            /// Deserializzazione xml
            /// </summary>
            /// <param name="fieldName"></param>
            /// <param name="fieldValue"></param>
            protected void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "IDRicontatto":
                        {
                            IDRicontatto = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "TipoRicontatto":
                        {
                            TipoRicontatto = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "IDContattoPrecedente":
                        {
                            IDContattoPrecedente = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "DataProssimoRicontatto":
                        {
                            DataProssimoRicontatto = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "DataProssimoAppuntamento":
                        {
                            DataProssimoAppuntamento = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "DataUltimoRicontatto":
                        {
                            DataUltimoRicontatto = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "DataUltimoAppuntamento":
                        {
                            DataUltimoAppuntamento = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "NumeroPraticheInCorso":
                        {
                            NumeroPraticheInCorso = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "NumeroPraticheArchiviate":
                        {
                            NumeroPraticheArchiviate = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }
                }
            }

            //~CCalendarActivityInfo()
            //{
            //    //DMDObject.DecreaseCounter(this);
            //}
        }
    }
}